orangecontrib/spectroscopy/widgets/owcos.py

```diff
@@ -417,18 +417,26 @@ def plotCOS(self):
             self.cbarCOS.set_range(-1 * np.nanmax(np.absolute(cosmat)), np.nanmax(np.absolute(cosmat)))
             self.cbarCOS.set_colors(np.array(colorcet.diverging_bwr_40_95_c42) * 255)
 
-            for s in leftSP:
-                self.left_plot.plot(s, leftSPwn, pen=pg.mkPen(color=(50, 50, 50), width=0.5))
+            left_indices = np.linspace(0, len(leftSP)-1, 100, dtype=int)
+            for s in leftSP[left_indices]:
+                pt = pg.PlotCurveItem(s, leftSPwn, pen=pg.mkPen(color=(50, 50, 50), width=0.5))
+                self.left_plot.addItem(pt, ignoreBounds=True)
 
             self.left_plot.plot(leftSP.mean(axis=0), leftSPwn, pen=p)
             self.left_plot.addItem(self.left_hLine)
 
-            for s in topSP:
-                self.top_plot.plot(topSPwn, s, pen=pg.mkPen(color=(50, 50, 50), width=0.5))
+            self.left_plot.setXRange(np.min(leftSP), np.max(leftSP))
+
+            top_indices = np.linspace(0, len(topSP)-1, 100, dtype=int)
```

Comment:
skip indices if you have <100 spectra, now this would repeat them.

Reply:
Good point, thanks!